spec/integration/api/messages_controller_spec.rb

```diff
@@ -92,12 +96,34 @@
         messages = JSON.parse(response.body)
         expect(messages.length).to eq 1
 
-        message = messages[0]
-        expect(message["guid"]).to_not be_nil
-        expect(message["author"]).to_not be_nil
-        expect(message["created_at"]).to_not be_nil
-        expect(message["body"]).to_not be_nil
+        confirm_message_format(messages[0], "first message", auth.user)
+        conversation = get_conversation(@conversation_guid)
+        expect(conversation[:read]).to be_truthy
       end
     end
   end
+
+  private
+
+  def get_conversation(conversation_id)
+    conversation_service = ConversationService.new(auth.user)
+    raw_conversation = conversation_service.find!(conversation_id)
+    ConversationPresenter.new(raw_conversation).as_api_json
+  end
+
+  def confirm_message_format(message, ref_message, author)
+    expect(message["guid"]).to_not be_nil
+    expect(message["created_at"]).to_not be_nil
+    expect(message["body"]).to eq ref_message
+    confirm_person_format(message["author"], author)
+  end
+
+  # rubocop:disable Metrics/AbcSize
+  def confirm_person_format(post_person, user)
```

Comment:
Maybe we could do a base class or a helpers module that contains confirm functions.

Reply:
I experimented with that but I couldn't get the RSpec stuff working with a "just pass in the type" per the RSPEC examples.  It's not like JUnit where you can write it in an aux library easily or my lack of Ruby experience is biting me on this.

Reply:
Ok